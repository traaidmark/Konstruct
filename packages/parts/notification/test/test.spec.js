// PART [ NOTIFICATION > TESTS ] ###############################################

// 1. DEPENDENCIES +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 1.1. EXTERNAL DEPENDENCIES ..................................................

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// 1.1. END ....................................................................

// 1.2. INTERNAL DEPENDENCIES ..................................................

import Props from './props';

import Notification from '../src';

// 1.2. END ....................................................................

// 1. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 2. TEST +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

describe('Part: Card', () => {

  // 2.1. RENDER WITHOUT PROPS .................................................

  test('renders correctly without any props', () => {
    const tree = renderer.create(
      <Notification />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 2.1. END ..................................................................

  // 2.2. RENDER WITH ALL PROPS ................................................

  test('renders correctly with all props', () => {
    const tree = renderer.create(
      <Notification { ...Props } />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 2.2. END ..................................................................

});

// 2. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// END OF FILE #################################################################
