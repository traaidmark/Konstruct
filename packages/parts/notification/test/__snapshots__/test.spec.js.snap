// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Part: Card renders correctly with all props 1`] = `
<aside
  className="p-notification "
>
  <figure
    className="p-notification__icon"
  >
    <img
      alt="notification icon"
      src="https://s3.amazonaws.com/assets.wealthpoint.app/images/icons/cat-risk-core.svg"
    />
  </figure>
  <p>
    In et sollicitudin dolor. Integer ut eleifend libero. Integer nec tellus a turpis porta posuere a non justo. Curabitur consequat sit amet augue ut aliquam. Quisque eu 
    <a>
      lobortis nibh
    </a>
    , vitae fringilla dolor.
  </p>
</aside>
`;

exports[`Part: Card renders correctly without any props 1`] = `
<aside
  className="p-notification "
/>
`;
